File: src/test/fixtures/kotlin/AnonymousCallableAnalysis.kt
val cleanAnonymousCallable = { first: Int, second: Int -> first + second }

val flaggedAnonymousCyclomatic = { value: Int ->
    if (value == 0) return@flaggedAnonymousCyclomatic 0
    if (value == 1) return@flaggedAnonymousCyclomatic 1
    if (value == 2) return@flaggedAnonymousCyclomatic 2
    if (value == 3) return@flaggedAnonymousCyclomatic 3
    if (value == 4) return@flaggedAnonymousCyclomatic 4
    if (value == 5) return@flaggedAnonymousCyclomatic 5
    if (value == 6) return@flaggedAnonymousCyclomatic 6
    if (value == 7) return@flaggedAnonymousCyclomatic 7
    if (value == 8) return@flaggedAnonymousCyclomatic 8
    if (value == 9) return@flaggedAnonymousCyclomatic 9
    if (value == 10) return@flaggedAnonymousCyclomatic 10
    value
}

val flaggedSevereAnonymousCyclomatic = { value: Int ->
    if (value == 0) return@flaggedSevereAnonymousCyclomatic 0
    if (value == 1) return@flaggedSevereAnonymousCyclomatic 1
    if (value == 2) return@flaggedSevereAnonymousCyclomatic 2
    if (value == 3) return@flaggedSevereAnonymousCyclomatic 3
    if (value == 4) return@flaggedSevereAnonymousCyclomatic 4
    if (value == 5) return@flaggedSevereAnonymousCyclomatic 5
    if (value == 6) return@flaggedSevereAnonymousCyclomatic 6
    if (value == 7) return@flaggedSevereAnonymousCyclomatic 7
    if (value == 8) return@flaggedSevereAnonymousCyclomatic 8
    if (value == 9) return@flaggedSevereAnonymousCyclomatic 9
    if (value == 10) return@flaggedSevereAnonymousCyclomatic 10
    if (value == 11) return@flaggedSevereAnonymousCyclomatic 11
    if (value == 12) return@flaggedSevereAnonymousCyclomatic 12
    if (value == 13) return@flaggedSevereAnonymousCyclomatic 13
    if (value == 14) return@flaggedSevereAnonymousCyclomatic 14
    if (value == 15) return@flaggedSevereAnonymousCyclomatic 15
    value
}

val flaggedAnonymousCognitive = { value: Int ->
    if (value > 0) {
        if (value > 1) {
            if (value > 2) {
                if (value > 3) {
                    if (value > 4) {
                        if (value > 5) value
                    }
                }
            }
        }
    }
    0
}

val flaggedSevereAnonymousCognitive = { value: Int ->
    if (value > 0) {
        if (value > 1) {
            if (value > 2) {
                if (value > 3) {
                    if (value > 4) {
                        if (value > 5) {
                            if (value > 6) value
                        }
                    }
                }
            }
        }
    }
    0
}

val flaggedAnonymousParameters = { a: Int, b: Int, c: Int, d: Int, e: Int, f: Int ->
    a + b + c + d + e + f
}

val flaggedSevereAnonymousParameters = { a: Int, b: Int, c: Int, d: Int, e: Int, f: Int, g: Int, h: Int, i: Int ->
    a + b + c + d + e + f + g + h + i
}

fun nestedCallableBoundary(value: Int): Int {
    val callback = { item: Int -> if (item > 0) 1 else 0 }
    return callback(value)
}

File: src/test/fixtures/csharp/AnonymousCallableAnalysis.cs
using System;

class CleanAnonymousCallable
{
    static Func<int, int, int> CleanAnonymousCallable = (first, second) => first + second;
}

class MediumAnonymousCyclomatic
{
    static Func<int, int> FlaggedAnonymousCyclomatic = value =>
    {
        if (value == 0) return 0;
        if (value == 1) return 1;
        if (value == 2) return 2;
        if (value == 3) return 3;
        if (value == 4) return 4;
        if (value == 5) return 5;
        if (value == 6) return 6;
        if (value == 7) return 7;
        if (value == 8) return 8;
        if (value == 9) return 9;
        if (value == 10) return 10;
        return value;
    };
}

class HighAnonymousCyclomatic
{
    static Func<int, int> FlaggedSevereAnonymousCyclomatic = value =>
    {
        if (value == 0) return 0;
        if (value == 1) return 1;
        if (value == 2) return 2;
        if (value == 3) return 3;
        if (value == 4) return 4;
        if (value == 5) return 5;
        if (value == 6) return 6;
        if (value == 7) return 7;
        if (value == 8) return 8;
        if (value == 9) return 9;
        if (value == 10) return 10;
        if (value == 11) return 11;
        if (value == 12) return 12;
        if (value == 13) return 13;
        if (value == 14) return 14;
        if (value == 15) return 15;
        return value;
    };
}

class MediumAnonymousCognitive
{
    static Func<int, int> FlaggedAnonymousCognitive = value =>
    {
        if (value > 0) {
            if (value > 1) {
                if (value > 2) {
                    if (value > 3) {
                        if (value > 4) {
                            if (value > 5) return value;
                        }
                    }
                }
            }
        }
        return 0;
    };
}

class HighAnonymousCognitive
{
    static Func<int, int> FlaggedSevereAnonymousCognitive = value =>
    {
        if (value > 0) {
            if (value > 1) {
                if (value > 2) {
                    if (value > 3) {
                        if (value > 4) {
                            if (value > 5) {
                                if (value > 6) return value;
                            }
                        }
                    }
                }
            }
        }
        return 0;
    };
}

class MediumAnonymousParameters
{
    delegate int SixParameters(int a, int b, int c, int d, int e, int f);
    static SixParameters FlaggedAnonymousParameters = delegate(int a, int b, int c, int d, int e, int f)
    {
        return a + b + c + d + e + f;
    };
}

class HighAnonymousParameters
{
    delegate int NineParameters(int a, int b, int c, int d, int e, int f, int g, int h, int i);
    static NineParameters FlaggedSevereAnonymousParameters =
        (a, b, c, d, e, f, g, h, i) => a + b + c + d + e + f + g + h + i;
}

class AnonymousParameterForms
{
    delegate int ParameterForms(int a, int b, int c, int d, int e, int f);
    static ParameterForms FlaggedAnonymousParameterForms =
        (int a, int b = 0, int c = 0, int d = 0, int e = 0, int f = 0) =>
            a + b + c + d + e + f;
}

class NestedCallableExample
{
    static int NestedCallableBoundary(int value)
    {
        Func<int, int> callback = item => item > 0 ? 1 : 0;
        return callback(value);
    }
}
